public/js/us-map.js

```diff
@@ -16,19 +16,80 @@ const tooltip = d3.select("body")
 		.append("div")
     .attr("class", "tooltip")
 
-const renderUS = (us) => {
+// Define scale for output
+var x = d3.scaleLinear()
+    .domain([1, 3])
+    .rangeRound([600, 860]);
+
+var color = d3.scaleThreshold()
+        .domain(d3.range(2, 5))
+        .range(d3.schemeGreys[3]);
+
+var g = svg.append("g")
+    .attr("class", "key")
+    .attr("transform", "translate(0,50)");
+
+const renderUS = (error, us, statePopulations) => {
   svg.append("g")
       .attr("class", "states")
       .selectAll("path")
       .data(topojson.feature(us, us.objects.states).features)
       .enter().append("path")
-      .attr("d", path);
+      .attr("d", path)
+      .attr("fill", (d) => {
+        if (statePopulations.find(state => state.state === d.properties.name)) {
+          const indentifiedState = statePopulations.find(state => state.state === d.properties.name)
+          const population = Number(indentifiedState.population.replace(/,/g,''))
+
+          return color(d.rate =
+            fetch(`http://localhost:3000/api/v1/state-territory/${nameToAbbreviation(d.properties.name)}/incidents`)
+              .then(response => response.json())
+              .then(response => {
+                  console.log(
```

Comment:
The range is working but I don't think this math is correct- I'm trying to get the rate of shootings per million people by multiplying the number of incidents (the length of the response) by 1,000,000 and then dividing that number by the state's population...  The results are console logging much larger than I think they should be?  @brittanystoroz